sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -357,24 +358,28 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
         /// A <see cref="RequestFailedException"/> will be thrown if
         /// a failure occurs.
         /// </remarks>
+        [ForwardsClientCalls]
```

Comment:
We shouldn't be using a `ForwardsClientCalls` attribute here.  Instead we should pass in the scope name to `PartitionedUploader` so it can switch between `$"{nameof(Azure)}.{nameof(Storage)}.{nameof(Blobs)}.{nameof(BlobClient)}.{nameof(BlobClient.Upload)}"` and `$"{nameof(Azure)}.{nameof(Storage)}.{nameof(Blobs)}.{nameof(Specialized)}.{nameof(BlockBlobClient)}.{nameof(BlockBlobClient.Upload)}"`.

Reply:
I can make this change, but keep in mind that BlobClient.Upload() also has [FowardsClientCalls]

Reply:
Oh - good catch.  I think `BlobClient.Upload` had leftover `[ForardsClientCalls]`s that should be removed now that `ParitionedUploader` is adding the correct scope.